test/16-fairness-tax.ts

```diff
@@ -0,0 +1,346 @@
+        expect(taxBPEarlier).to.equal(100)
+
+        await cartographerSetParam.setTokenWithdrawTax({
+            dev,
+            tokenAddress: cakeToken.address,
+            taxBP: 500
+        })
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: cakeToken.address,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+
+        const afterDepositTimestamp = await getTimestamp()
+        await mineBlockWithTimestamp(afterDepositTimestamp + days(7))
+
+        const taxBPAfter = await cartographerGet.taxBP(user1.address, cakeToken.address)
+        expect(taxBPAfter).to.equal(100)
+    })
+
+    it(`WAIVED TAX: The fairness tax is waived when a use elevates their SUMMIT from the elevation farms to the Expedition`, async function() {
+        const { user1 } = await getNamedSigners(hre)
+        const cakeToken = await getSummitToken()
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: cakeToken.address,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+        await everestMethod.lockSummit({
+            user: user1,
+            amount: e18(1),
+            lockDuration: days(7),
+        })
+
+        const userTotalStakedAmountBefore = await cartographerGet.userTokenStakedAmount(user1.address, cakeToken.address)
+        await cartographerMethod.elevateAndLockStakedSummit({
+            user: user1,
+            elevation: OASIS,
+            amount: e18(5),
+        })
+        const userTotalStakedAmountAfter = await cartographerGet.userTokenStakedAmount(user1.address, cakeToken.address)
+
+        expect(userTotalStakedAmountAfter + 5).to.equal(userTotalStakedAmountBefore)
+    })
+
+    it(`MINIMUM TAX: The 0% tax should be taken when a use withdraws from a farm after the tax has fully decayed (for native farms) `, async function() {
+        const { user1 } = await getNamedSigners(hre)
+        const summitToken = await getSummitToken()
+        const depositAmount = 5
+        const withdrawAmount = 5
+
+        await cartographerMethod.deposit({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: OASIS,
+            amount: e18(depositAmount),
+        })    
+
+        const afterDepositTimestamp = await getTimestamp()
+        await mineBlockWithTimestamp(afterDepositTimestamp + days(8))
+
+        const summitTokenBalanceBefore = (await summitToken.balanceOf(user1.address)).toString()
+        await cartographerMethod.withdraw({
+            user: user1,
+            tokenAddress: summitToken.address,
+            elevation: OASIS,
+            amount: e18(withdrawAmount),
+        })
+        const summitTokenBalanceAfter = (await summitToken.balanceOf(user1.address)).toString()
+        expect(Number(summitTokenBalanceAfter)).to.equal(Number(summitTokenBalanceBefore) + Number(e18(withdrawAmount)))
```

Comment:
there is a utility function `expectBigNumbersEqual` that you can use that doesn't require you to cast to a number to do the comparison. Also the built in expect works for BigNumbers as well